src/high-level/scripts/papi_hl_output_writer.py

```diff
@@ -471,45 +506,71 @@ def write_json_file(data, file_name):
     outfile.write(to_unicode(str_))
     print (str_)
 
+def main(format, type, notation, source = None, source_file = None):
+    if (format == "json"):
+        if source != None:
+            json = merge_json_files(source)
+        else:
+            json = parse_source_file(source_file)
 
-def main(source, format, type, notation):
-  if (format == "json"):
-    json = merge_json_files(source)
-
-    if type == 'detail':
-      if notation == 'derived':
-        write_json_file(derive_json_object(json), 'papi.json')
-      else:
-        write_json_file(json, 'papi.json')
-
-    #summarize data over regions with the same name, threads and ranks
-    if type == 'summary':
-      if notation == 'derived':
-        write_json_file(sum_json_object(json, True), 'papi_sum.json')
-      else:
-        write_json_file(sum_json_object(json), 'papi_sum.json')
+        if type == 'detail':
+            if notation == 'derived':
+                write_json_file(derive_json_object(json), 'papi.json')
+            else:
+                write_json_file(json, 'papi.json')
 
-  else:
-    print("Format not supported!")
+        #summarize data over regions with the same name, threads and ranks
+        if type == 'summary':
+            if notation == 'derived':
+                write_json_file(sum_json_object(json, True), 'papi_sum.json')
+            else:
+                write_json_file(sum_json_object(json), 'papi_sum.json')
+    else:
+        print("Format not supported!")
 
 
 def parse_args():
   parser = argparse.ArgumentParser()
-  parser.add_argument('--source', type=str, required=False, default="papi_hl_output",
```

Comment:
One suggestion is we could change `--source` to `--source_dir`. This would add more distinction to what a user is passing and it would keep inline with `--source_file`.  Nothing major, and we definitely could keep it as `source` if preferred.

Reply:
Changed `--source` to `--source_dir` in latest commit.